test/compress/destructuring.js

```diff
@@ -639,3 +639,25 @@ issue_2044_ecma_6_beautify: {
     }
     expect_exact: "({x: a = 1, y = 2 + b, z = 3 - c} = obj);"
 }
+
+issue_2140: {
+    options = {
+        unused: true,
+    }
+    input: {
+        !function() {
+            var s = {}, t = {};
+            [t.a] = [42];
+            console.log(t.a);
```

Comment:
@alexlamsl Are you sure this test case failed before? It doesn't seem to fail for me before this PR is applied.

Could you please add the test from https://github.com/mishoo/UglifyJS2/issues/2140#issuecomment-310285807 exactly as posted?